core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -256,9 +254,10 @@ class LogManager(logDirs: Seq[File],
   private[log] def loadLog(logDir: File,
                       hadCleanShutdown: Boolean,
                       recoveryPoints: Map[TopicPartition, Long],
-                      logStartOffsets: Map[TopicPartition, Long]): Log = {
+                      logStartOffsets: Map[TopicPartition, Long],
+                      topicLogConfigs: Map[String, LogConfig]): Log = {
     val topicPartition = Log.parseTopicPartitionName(logDir)
-    val config = topicConfigs.getOrElse(topicPartition.topic, currentDefaultConfig)
+    val config = topicLogConfigs.getOrElse(topicPartition.topic, currentDefaultConfig)
```

Comment:
We can ignore any topic that has no explicit configs in `startup()` below since this is doing `.getOrElse()` -- it'll save memory.